riscv/src/riscv_gl.asm

```diff
@@ -0,0 +1,345 @@
+{{IMPORTS}}
+
+let MIN_DEGREE_LOG: int = {{MIN_DEGREE_LOG}};
+let MIN_DEGREE: int = 2**MIN_DEGREE_LOG;
+let MAX_DEGREE_LOG: int = {{MAX_DEGREE_LOG}};
+let MAIN_MAX_DEGREE: int = 2**MAX_DEGREE_LOG;
+let LARGE_SUBMACHINES_MAX_DEGREE: int = 2**(MAX_DEGREE_LOG + 2);
+
+machine Main with min_degree: MIN_DEGREE, max_degree: {{MAIN_MAX_DEGREE}} {
+
+    reg pc[@pc];
+    reg X[<=];
+    reg Y[<=];
+    reg Z[<=];
+    reg W[<=];
+    {{SUBMACHINE_EXTRA_REGISTERS}}
+
+    // We still need these registers for prover inputs.
+    reg query_arg_1;
+    reg query_arg_2;
+
+    std::machines::large_field::memory::Memory regs(byte2, MIN_DEGREE, LARGE_SUBMACHINES_MAX_DEGREE);
+    {{MEMORY_DECLARATION}}
+    {{SUBMACHINE_DECLARATIONS}}
+
+    let initial_memory: (fe, fe)[] = [
+        {{INITIAL_MEMORY}}
+    ];
+
+    function main {
+        {{PROGRAM}}
+    }
+
+    // Extra columns we use to hold temporary values inside instructions.
```

Comment:
the `col` declarations are a bit scattered around the file, maybe they could be placed together? like the `tmpX_col` ones,  `Y_7bit` etc

Reply:
I copied everything mostly as is from the original file, trying not to mess with the template too much to avoid diff hell, but looks like github didn't care anyway

Reply:
done